Foundation/HTTPCookie.swift

```diff
@@ -444,14 +488,16 @@ open class HTTPCookie : NSObject {
     }
 
     //Bake a cookie
-    private class func createHttpCookie(url: URL, pairs: ArraySlice<String>) -> HTTPCookie? {
-        var properties: [HTTPCookiePropertyKey : Any] = [:]
+    private class func createHttpCookie(url: URL, cookie: Substring) -> HTTPCookie? {
+        let pairs = cookie.components(separatedBy: ";") // split the name/value and attribute/value pairs
+                          .map { $0.trim() }            // trim whitespaces
+                          .filter { $0.caseInsensitiveCompare("HTTPOnly") != .orderedSame }  // we don't use HTTPOnly, do we?
+                          .flatMap(HTTPCookie.createNameValuePair) // create Name and Value properties
+
+        var properties: [HTTPCookiePropertyKey : String] = [:]
         for pair in pairs {
             let name = pair.components(separatedBy: "=")[0]
-            var value = pair.components(separatedBy: "\(name)=")[1]  //a value can have an "="
-            if canonicalize(name) == .expires {
-                value = value.unmaskCommas()    //re-insert the comma
-            }
+            let value = pair.components(separatedBy: "\(name)=")[1]  //a value can have an "="
```

Comment:
Is the string guaranteed to have an `=` in it? If not this will cause an array out of bounds error. It might be safer to write this as:
```
if let idx1 = pair.firstIndex(of: "=")  {
        let idx2 = pair.index(after: idx1)
        let name = String(pair[..<idx1])
        let value = String(pair[idx2...])
        properties[canonicalize(name)] = value
}
```

With checks for `name.isEmpty` as appropriate. This should also be faster than calling `.components(separatedBy:)` twice.

Reply:
I think since these is strings that are sent from the server, one cannot have any guarantee.

My intention with these changes was fixing the cookie split algorithm, but you are right that the cookie parsing algorith also needs some love (for example you point out that the name/value split seems to be happening in two points). I think much of the previous algorithm relied on crashing on badly formatted input (for example if there's no `=`, there's no `[1]` to get the value).

I was planning to improve the cookie parsing in a later commit, making it closer to the grammar in the RFC, but I was planning to do it in a separate PR. Would that be good?

Reply:
If you want to do it in a follow-up PR and merge the two places thats fine.